trio/_tools/gen_exports.py

```diff
@@ -104,6 +105,40 @@ def create_passthrough_args(funcdef: ast.FunctionDef | ast.AsyncFunctionDef) ->
     return "({})".format(", ".join(call_args))
 
 
+def run_linters(file: File, source: str) -> str:
+    """Run isort and black on the specified file, returning the new source.
+
+    :raises ValueError: If either failed.
+    """
+    # Black has an undocumented API, but it doesn't easily allow reading configuration from
+    # pyproject.toml, and simultaneously pass in / receive the code as a string.
+    # https://github.com/psf/black/issues/779
+    try:
+        result = subprocess.run(
+            # "-" as a filename = use stdin, return on stdout.
+            [sys.executable, "-m", "black", "--stdin-filename", file.path, "-"],
+            input=source,
+            capture_output=True,
+            encoding="utf8",
+            check=True,
+        )
+    except subprocess.CalledProcessError as exc:
+        print("Failed to run black!")
+        traceback.print_exception(type(exc), exc, exc.__traceback__)
+        sys.exit(1)
+    # isort does have a public API, makes things easy.
+    try:
+        isort_res = isort.api.sort_code_string(
+            result.stdout,
+            file_path=file.path,
+        )
+    except isort.exceptions.ISortError as exc:
+        print("Failed to run isort!")
+        traceback.print_exception(type(exc), exc, exc.__traceback__)
+        sys.exit(1)
```

Comment:
Why can't we re-raise? Or even `raise RuntimeError("Failed to run isort!") from exc`!

Reply:
I don't think that results in a nonzero exit code, I wanted to make sure CI etc fails.

Reply:
FWIW it does, it seems!

```py
a5@hostname:/tmp$ python3 t.py
Traceback (most recent call last):
  File "t.py", line 1, in <module>
    raise Exception("hi!")
Exception: hi!
a5@hostname:/tmp$ echo $?
1
```
(at least on my Ubuntu WSL env...)